### PR TITLE
OpenAPI: Fix additionalProperties for SnapshotSummary

### PR DESCRIPTION
**Pull request by haizhou-zhao**
Details described on https://github.com/apache/iceberg/issues/9837

### CODE REVIEW

**Comment by dramaticlly** [comment]
Can you do make generate to have python code update as well? Otherwise LGTM

**Comment by haizhou-zhao** [comment]
> Can you do make generate to have python code update as well? Otherwise LGTM

@dramaticlly Tested with OpenAPI's python generator (if that's what you mean), details also in the issue https://github.com/apache/iceberg/issues/9837. Though you'll need to scroll a bit down to see that.

**Comment by dramaticlly** [comment]
> > Can you do make generate to have python code update as well? Otherwise LGTM
> 
> 
> 
> @dramaticlly Tested with OpenAPI's python generator (if that's what you mean), details also in the issue https://github.com/apache/iceberg/issues/9837. Though you'll need to scroll a bit down to see that.

Can you check if your workspace is empty after generate? I think it will update the python file and you need that as part of your commit. You can see pull request build also failed for the same reason

**Comment by dramaticlly** [comment]
@nastra @Fokko can you help review the change?

File: open-api/rest-catalog-open-api.py [hunk 1]
```diff
@@ -171,7 +171,6 @@ class SortOrder(BaseModel):
 
 class Summary(BaseModel):
     operation: Literal['append', 'replace', 'overwrite', 'delete']
-    additionalProperties: Optional[str] = None
```
Comment (nastra) [inline open-api/rest-catalog-open-api.py:174]: I would have expected the summary to also have `Optional[Dict[str, str]] = None`

File: open-api/rest-catalog-open-api.yaml [hunk 2]
```diff
@@ -1972,8 +1972,8 @@ components:
             operation:
               type: string
               enum: ["append", "replace", "overwrite", "delete"]
-            additionalProperties:
-              type: string
+          additionalProperties:
```
Comment (nastra) [inline open-api/rest-catalog-open-api.yaml:1975]: this change LGTM and also conforms to https://swagger.io/docs/specification/data-models/dictionaries/

**Review by nastra — APPROVED** [review]
LGTM, @Fokko could you please take a look at the generated python code?

File: open-api/rest-catalog-open-api.py (hunk 1 above)
Comment (haizhou-zhao) [reply open-api/rest-catalog-open-api.py:174]: When I used the plain OpenAPI generator (tested on 6.6.0 & 7.0.0) to generate python code, this is what I got
```
...
class SnapshotSummary(BaseModel):
    """
    SnapshotSummary
    """
    operation: StrictStr = Field(...)
    additional_properties: Dict[str, Any] = {}
    __properties = ["operation"]
...
```
Which is closer to what you suggested above. I guess the code generation process here might leverage some different library.

**Review by Fokko — APPROVED** [review]
Looks good. Thanks @haizhou-zhao for working on this, and @nastra for the review 👍

**Pull request merged**